osu.Game/Overlays/FirstRunSetup/ScreenImportFromStable.cs

```diff
@@ -0,0 +1,268 @@
+{
+    [LocalisableDescription(typeof(FirstRunOverlayImportFromStableScreenStrings), nameof(FirstRunOverlayImportFromStableScreenStrings.Header))]
+    public class ScreenImportFromStable : FirstRunSetupScreen
+    {
+        private static readonly Vector2 button_size = new Vector2(400, 50);
+
+        private ProgressRoundedButton importButton = null!;
+
+        private OsuTextFlowContainer progressText = null!;
+
+        [Resolved]
+        private LegacyImportManager legacyImportManager { get; set; } = null!;
+
+        private StableLocatorLabelledTextBox stableLocatorTextBox = null!;
+
+        private IEnumerable<ImportCheckbox> contentCheckboxes => Content.Children.OfType<ImportCheckbox>();
+
+        [BackgroundDependencyLoader(permitNulls: true)]
+        private void load()
+        {
+            Content.Children = new Drawable[]
+            {
+                new OsuTextFlowContainer(cp => cp.Font = OsuFont.Default.With(size: CONTENT_FONT_SIZE))
+                {
+                    Colour = OverlayColourProvider.Content1,
+                    Text = FirstRunOverlayImportFromStableScreenStrings.Description,
+                    RelativeSizeAxes = Axes.X,
+                    AutoSizeAxes = Axes.Y
+                },
+                stableLocatorTextBox = new StableLocatorLabelledTextBox
+                {
+                    Label = FirstRunOverlayImportFromStableScreenStrings.LocateDirectoryLabel,
+                    PlaceholderText = FirstRunOverlayImportFromStableScreenStrings.LocateDirectoryPlaceholder
+                },
+                new ImportCheckbox("Beatmaps", StableContent.Beatmaps),
+                new ImportCheckbox("Scores", StableContent.Scores),
+                new ImportCheckbox("Skins", StableContent.Skins),
+                new ImportCheckbox("Collections", StableContent.Collections),
+                importButton = new ProgressRoundedButton
+                {
+                    Size = button_size,
+                    Anchor = Anchor.TopCentre,
+                    Origin = Anchor.TopCentre,
+                    Text = FirstRunOverlayImportFromStableScreenStrings.ImportButton,
+                    Action = runImport
+                },
+                progressText = new OsuTextFlowContainer(cp => cp.Font = OsuFont.Default.With(size: CONTENT_FONT_SIZE))
+                {
+                    Colour = OverlayColourProvider.Content1,
+                    Text = FirstRunOverlayImportFromStableScreenStrings.ImportInProgress,
+                    RelativeSizeAxes = Axes.X,
+                    AutoSizeAxes = Axes.Y,
+                    Alpha = 0,
+                },
+            };
+
+            stableLocatorTextBox.Current.BindValueChanged(_ => updateStablePath(), true);
+        }
+
+        private void updateStablePath()
+        {
+            var storage = legacyImportManager.GetCurrentStableStorage();
+
+            if (storage == null)
+            {
+                allowInteraction(false);
```

Comment:
Not sure how to feel about an `allowInteraction(false)` call... disabling interaction. Maybe `toggleInteraction(false)` at least.